compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ExpectedTypeFinder.java

```diff
@@ -585,7 +605,10 @@ public Optional<TypeSymbol> transform(RemoteMethodCallActionNode node) {
         if (bLangNode instanceof BLangInvocation.BLangActionInvocation) {
             BLangInvocation.BLangActionInvocation bLangActionInvocation = (BLangInvocation.BLangActionInvocation)
                     nodeFinder.lookup(this.bLangCompilationUnit, node.lineRange());
-            if (isWithinParenthesis(linePosition, node.openParenToken(), node.closeParenToken())) {
+            Token openParen = node.openParenToken();
+            Token closeParen = node.closeParenToken();
+            if (!isParenthesisMissing(openParen, closeParen) &&
+                    isWithinParenthesis(linePosition, openParen.lineRange(), closeParen.lineRange())) {
```

Comment:
If we are calling these two multiple times as a whole, we can combine them and extract to a function